presto-main/src/main/java/io/prestosql/execution/CreateTableTask.java

```diff
@@ -154,14 +178,22 @@ public String explain(CreateTable statement, List<Expression> parameters)
             else if (element instanceof LikeClause) {
                 LikeClause likeClause = (LikeClause) element;
                 QualifiedObjectName likeTableName = createQualifiedObjectName(session, statement, likeClause.getTableName());
+                Optional<QualifiedObjectName> redirectedLikeTableName = metadata.redirectTable(session, likeTableName, true);
+                if (redirectedLikeTableName.isPresent()) {
+                    likeTableName = redirectedLikeTableName.get();
+                    warningCollector.add(new PrestoWarning(REDIRECTED_TABLE, "Table redirection happened"));
+                }
                 if (metadata.getCatalogHandle(session, likeTableName.getCatalogName()).isEmpty()) {
                     throw semanticException(CATALOG_NOT_FOUND, statement, "LIKE table catalog '%s' does not exist", likeTableName.getCatalogName());
                 }
                 if (!tableName.getCatalogName().equals(likeTableName.getCatalogName())) {
                     throw semanticException(NOT_SUPPORTED, statement, "LIKE table across catalogs is not supported");
                 }
-                TableHandle likeTable = metadata.getTableHandle(session, likeTableName)
```

Comment:
It doesn't look like this code needs to change. I can't see what is different with regards to redirection.